### ISSUE TITLE
Problem with defines called out of order

### ISSUE THREAD

**Issue by moschel** [bitovi/canjs#1519]
Defines that are dependent on each other seem to not fire in the correct order. I have a breaking test showing this here: https://github.com/bitovi/canjs/commit/a3d9d38915e5425808d94b677a092f5a0e7cb22f

**Comment by moschel** [comment]
Here's the breaking test:

````js
		var MyMap = can.Map.extend({
		  define: {
		    fulfillmentType: {
		      get: function() {
		        return this.attr('ft');
		      },

		      set: function(newVal) {
		        this.attr('ft', newVal);
		        return newVal;
		      }
		    },

		    propA: {
		      get: function() {
		        return this.attr('fulfillmentType') == 'a';
		      }
		    },

		    propB: {
		      get: function() {
		        return this.attr('fulfillmentType') == 'b';
		      }
		    },
		    
		    combined: {
		      get: function() {
		        return this.attr('propA') || this.attr('propB');
		      }
		    }
		  }
		});

		var map = new MyMap({ft: 'a'});

		map.on('combined', function(){});
		map.on('propA', function(){});
		map.on('propB', function(){});

		map.attr('fulfillmentType', 'b');
		equal(map.attr('combined'), true);
````

This seems very dependent on the order. If I change the order the properties are bound, it doesn't break. If I remove propA or propB altogether, it doesn't break.

**Comment by daffl** [comment]
This is closed via #1520

**Comment by m-mujica** [comment]
It seems that this issue was not totally fixed. In [this branch](https://github.com/bitovi/canjs/tree/bug-nested-computes) I've created a new failing test with more complex computes. 

![nested-computes](https://cloud.githubusercontent.com/assets/724877/7068077/47f5b7b8-dea6-11e4-8683-319eb8ed52ce.png)

I've also created a [jsbin](http://jsbin.com/limobe/2/edit?js,console) with application like code that this issue affects.
